be/src/olap/tablet.cpp

```diff
@@ -970,4 +878,42 @@ void Tablet::pick_candicate_rowsets_to_base_compaction(std::vector<RowsetSharedP
     }
 }
 
+OLAPStatus Tablet::capture_unused_rowsets() {
```

Comment:
You will iterate all tablet every time with lock, and I think it can be improved.